### ISSUE TITLE
Training code

### ISSUE THREAD

**Issue by yukariin** [hughplay/DFNet#1]
Training code is necessary to understand what loss functions were used, what was the model architecture etc. It would be a nice idea to add training code too.

**Comment by zhaowenlovehome** [comment]
That‘s all right

**Comment by hughplay** [comment]
Thanks for your attention. @Yukariin @zhaowenlovehome 
Our training code depend on a self-written framework and is a little bit mess now. I will rewrite the training code when I get some time.

I will release loss functions first if you want to try training by yourself.

Pull requests are also welcome.

**Comment by yukariin** [comment]
> I will release loss functions first if you want to try training by yourself.

Oh, that would be nice.

**Comment by shgidi** [comment]
Hi @hughplay , any updates re the training code?

**Comment by tlliao** [comment]
> Hi @hughplay , any updates re the training code?

I have the same problem.

**Comment by hughplay** [comment]
Please refer to https://github.com/deepcodebase/inpaint. It is building in progress but looks good so far.
